lib/SILOptimizer/Transforms/PhiArgumentOptimizations.cpp

```diff
@@ -157,7 +179,18 @@ bool RedundantPhiEliminationPass::valuesAreEqual(SILValue val1, SILValue val2) {
         return false;
       if (arg1->getType() != arg2->getType())
         return false;
-    
+
+      // Don't optimize if we have a phi arg with @guaranteed ownership.
+      // Such a phi arg will be redundant only if all the incoming values have
+      // none ownership.
```

Comment:
This is fine. But do we even allow `@none` to `@guaranteed` conversion? If so, I need to go fix some of my code to handle that case.